src/libcharon/plugins/kernel_super/kernel_super_ipsec.c

```diff
@@ -3796,6 +3804,48 @@ CALLBACK(terminator_tunnels_oper_timer_cb, job_requeue_t, private_kernel_super_i
 	return JOB_RESCHEDULE_MS(TERMINATORS_COUNTERS_TIMER_MS);
 }
 
+static bool init_system_events(private_kernel_super_ipsec_t *this)
+{
+	const char *system_events_type =
+		lib->settings->get_str(lib->settings, "%s.plugins.kernel-super.system_events_type", NULL,
+							   lib->ns);
+	system_events_t *system_events_interface = NULL;
+	LOG(DBG_KNL, SL_LOG_INFO, "using system_events_type '%s'", system_events_type);
+
+	if (streq("libsystem_events", system_events_type))
+	{
+		sl_system_events_t *const sl_system_events = sl_system_events_create();
+		if (! sl_system_events)
+		{
+			LOG(DBG_KNL, SL_LOG_ERR, "failed to create sl_system_events_t");
+			goto fail;
+		}
+		system_events_interface = &sl_system_events->interface;
+	}
+	else if (streq("empty", system_events_type))
+	{
+		empty_system_events_t *const empty_system_events = empty_system_events_create();
+		if (! empty_system_events)
+		{
+			LOG(DBG_KNL, SL_LOG_ERR, "failed to create empty_system_events_t");
+			goto fail;
+		}
+		system_events_interface = &empty_system_events->interface;
+	}
+	else
+	{
+		LOG(DBG_KNL, SL_LOG_ERR, "wrong system_events_type: %s", system_events_type);
+		goto fail;
+	}
+
+	this->system_events = system_events_interface;
+	return true;
+fail:
```

Comment:
Add empty line between regular flows and error goto labels
```suggestion
	return true;

fail:
```